### PR TITLE
Feature/david/fdroid compliance

### PR DESCRIPTION
**Pull request by malmstein**
Task/Issue URL:  https://app.asana.com/0/1157893581871903/1199944591737638

**Description**:
F-Droid does not allow the use of closed source libraries in the apps published in their store. Somehow, our use of Google Referrer Library had not been flagged.

This PR creates a new variant for Fdroid, where we don't use the Referrer Library but a Stub version of it. Doing this change will create two different release apks, one for fdroid and another one for play. 

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -178,7 +194,7 @@ dependencies {
     implementation "com.airbnb.android:lottie:_"
 
     // Play Store referrer library
-    implementation("com.android.installreferrer:installreferrer:_")
+    playImplementation("com.android.installreferrer:installreferrer:_")
```
Comment (malmstein) [inline app/build.gradle:197]: this dependency should only be available for the Play Store version

File: app/src/fdroid/java/com/duckduckgo/app/di/StoreReferralModule.kt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import com.duckduckgo.app.referral.*
+import com.duckduckgo.app.statistics.AtbInitializerListener
+import com.duckduckgo.referral.FDroidReferrerStateListener
+import dagger.Module
+import dagger.Provides
+import dagger.multibindings.IntoSet
+import javax.inject.Singleton
+
+@Module
+class StoreReferralModule {
```
Comment (malmstein) [inline app/src/fdroid/java/com/duckduckgo/app/di/StoreReferralModule.kt:29]: we have now two `StoreRefferralModule`, only one will be used based on the variant being built

File: app/src/main/java/com/duckduckgo/app/referral/AppInstallationReferrerStateListener.kt [hunk 3]
```diff
@@ -45,151 +26,3 @@ interface AppInstallationReferrerStateListener {
     }
 
 }
-
-@Singleton
```
Comment (malmstein) [inline app/src/main/java/com/duckduckgo/app/referral/AppInstallationReferrerStateListener.kt:49]: we can keep the `AppInstallationReferrerStateListener ` in the `main` folder, and each one of the variants will use its own implementation of the interface

File: fastlane/Fastfile [hunk 4]
```diff
@@ -22,7 +22,7 @@ platform :android do
 
     props = property_file_read(file: "app/version/version.properties")
     version = props["VERSION"]
-    apkPath = "app/build/outputs/apk/release/duckduckgo-#{version}-release.apk"
+    apkPath = "app/build/outputs/apk/play/release/duckduckgo-#{version}-play-release.apk"
```
Comment (malmstein) [inline fastlane/Fastfile:25]: update apk file path for uploading to Github

File: fastlane/Fastfile [hunk 5]
```diff
@@ -36,7 +36,7 @@ platform :android do
     props = property_file_read(file: "app/version/version.properties")
     version = props["VERSION"]
     releaseNotes = File.read("../app/version/release-notes")
-    apkPath = "app/build/outputs/apk/release/duckduckgo-#{version}-release.apk"
+    apkPath = "app/build/outputs/apk/play/release/duckduckgo-#{version}-play-release.apk"
```
Comment (malmstein) [inline fastlane/Fastfile:39]: update apk file path for uploading to Play Store

**Review by aitorvs — APPROVED** [review]
Nice, LGTM! 🚢 🇮🇹

**Pull request merged**